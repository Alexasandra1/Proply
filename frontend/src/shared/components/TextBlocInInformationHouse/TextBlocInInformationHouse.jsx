import React from 'react';
import './TextBlocInInformationHouse.scss';

export function TextBlocInInformationHouse(props) {
  return (
    <div className={`textBlocInInformationHouse__block ${props.styleTextBlocInInformationHouse}`}>
      <h1 className={`textBlocInInformationHouse__block__type`} >{props.typeStyleTextBlocInInformationHouse}</h1>
      <p className={`textBlocInInformationHouse__block__price`}>{props.priceStyleTextBlocInInformationHouse}</p>
      <p className={`textBlocInInformationHouse__block__description`}>{props.descriptionStyleTextBlocInInformationHouse}</p>
      <div className={`textBlocInInformationHouse__block__container`} >
        <p className={`textBlocInInformationHouse__block__container__carSpace`}>Car places:{props.carSpaceStyleTextBlocInInformationHouse}</p>
        <p className={`textBlocInInformationHouse__block__container__bedrooms`}>Bedrooms:{props.bedroomsStyleTextBlocInInformationHouse}</p>
        <p className={`textBlocInInformationHouse__block__container__bathrooms`}>Bathrooms:{props.bathroomsStyleTextBlocInInformationHouse}</p>
      </div>
      <p className={`textBlocInInformationHouse__block__adress`}>{props.adressStyleTextBlocInInformationHouse}</p>
    </div>
  );
}

