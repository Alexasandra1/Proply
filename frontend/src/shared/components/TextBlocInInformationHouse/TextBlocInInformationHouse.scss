// .textBlocInInformationDesign__block {
//     color: #FFF;
//     font-family: Montserrat;
//     font-weight: bolder;
//     font-size: 32px;

//     p{
//         text-align: left;
//     }

//     .textBlocInInformationDesign__block__price {
//         text-align: center;
//         font-size: 64px;
//     }
// }

.textBlocInInformationHouse__block__container {
    display: flex;
    flex-direction: row;
    gap: 2%
}

.textBlocInInformationHouse__block__type {
    color: #000;
    font-family: Montserrat;
    font-size: 36px;
}

.textBlocInInformationHouse__block__price {
    color: #2D7C83;
    font-family: Montserrat;
    font-size: 64px;
}

.textBlocInInformationHouse__block__description {
    color: #000;
    font-family: Montserrat;
    font-size: 16px;
}

.textBlocInInformationHouse__block__container__carSpace,
.textBlocInInformationHouse__block__container__bedrooms,
.textBlocInInformationHouse__block__container__bathrooms {
    color: #000;
    font-family: Montserrat;
    font-size: 20px;
}

.textBlocInInformationHouse__block__adress {
    color: #000;
    font-family: Montserrat;
    font-size: 18px;
}

@media (max-width: 949px) {
    .textBlocInInformationHouse__block__type,
    .textBlocInInformationHouse__block__price,
    .textBlocInInformationHouse__block__description,
    .textBlocInInformationHouse__block__adress {
        text-align: center;
    }
    .textBlocInInformationHouse__block__container{
        justify-content: center;
    }
    
}