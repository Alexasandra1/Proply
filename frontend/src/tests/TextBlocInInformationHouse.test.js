import React from 'react';
import renderer from 'react-test-renderer';
import { TextBlocInInformationHouse } from '../shared/components/TextBlocInInformationHouse/TextBlocInInformationHouse';

test('renders correctly', () => {
  const props = {
    styleTextBlocInInformationHouse: 'test-style',
    typeStyleTextBlocInInformationHouse: 'Test Type',
    priceStyleTextBlocInInformationHouse: '1000',
    descriptionStyleTextBlocInInformationHouse: 'Test Description',
    carSpaceStyleTextBlocInInformationHouse: '2',
    bedroomsStyleTextBlocInInformationHouse: '3',
    bathroomsStyleTextBlocInInformationHouse: '1',
    adressStyleTextBlocInInformationHouse: 'Test Address',
  };

  const tree = renderer
    .create(<TextBlocInInformationHouse {...props} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
