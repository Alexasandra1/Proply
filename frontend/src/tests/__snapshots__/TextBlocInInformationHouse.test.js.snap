// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="textBlocInInformationHouse__block test-style"
>
  <h1
    className="textBlocInInformationHouse__block__type"
  >
    Test Type
  </h1>
  <p
    className="textBlocInInformationHouse__block__price"
  >
    1000
  </p>
  <p
    className="textBlocInInformationHouse__block__description"
  >
    Test Description
  </p>
  <div
    className="textBlocInInformationHouse__block__container"
  >
    <p
      className="textBlocInInformationHouse__block__container__carSpace"
    >
      Car places:
      2
    </p>
    <p
      className="textBlocInInformationHouse__block__container__bedrooms"
    >
      Bedrooms:
      3
    </p>
    <p
      className="textBlocInInformationHouse__block__container__bathrooms"
    >
      Bathrooms:
      1
    </p>
  </div>
  <p
    className="textBlocInInformationHouse__block__adress"
  >
    Test Address
  </p>
</div>
`;
